### DISCUSSION TITLE
CustomLLM is missing

### DISCUSSION THREAD

**Discussion by eranitach** [run-llama/LlamaIndexTS#1584] [Q&A]
**Describe the bug**
Trying to implement CustomLLM , with different apiKey and baseUrl. 
Below you can see an example in Python.

https://docs.llamaindex.ai/en/stable/api_reference/llms/custom_llm/

**To Reproduce**
No need

**Expected behavior**
Option to use a custom LLM like in Python version

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. macOS, Linux]
- JS Runtime / Framework / Bundler (select all applicable)
- [ ] Node.js
- [ ] Deno
- [ ] Bun
- [ ] Next.js
- [ ] ESBuild
- [ ] Rollup
- [ ] Webpack
- [ ] Turbopack
- [ ] Vite
- [ ] Waku
- [ ] Edge Runtime
- [ ] AWS Lambda
- [ ] Cloudflare Worker
- [ ] Others (please elaborate on this)

**Additional context**

**Comment by marcusschiesser** [comment]
Which LLM do you want to add support for? To add a new LLM you'll need to add a new provider package here https://github.com/run-llama/LlamaIndexTS/tree/main/packages/providers
The one for Ollama is a good starter: https://github.com/run-llama/LlamaIndexTS/blob/main/packages/providers/ollama/src/llm.ts

**Reply by eranitach** [reply]
hi, thanks for the answer.

I want to add custom one, in my company we're working with our designated LLM, not one of the providers that you mentioned. I want to add something that isn't in the list. In Python we can do it (look at the link above). I don't see such example in TS..
